source/ed/ed_reader.cpp

```diff
@@ -199,6 +200,60 @@ void EdReader::fill_admin_stop_areas(navitia::type::Data&, pqxx::work& work) {
 
 }
 
+void EdReader::fill_object_codes(navitia::type::Data&, pqxx::work& work){
+    std::string request = "select object_type_id, object_id, key, value from object_code";
+
+    pqxx::result result = work.exec(request);
+    for(auto const_it = result.begin(); const_it != result.end(); ++const_it){
+        nt::Type_e object_type_e = static_cast<nt::Type_e>(const_it["object_type_id"].as<int>());
+        switch(object_type_e) {
+        case nt::Type_e::Line:{
+            auto it_object = this->line_map.find(const_it["object_id"].as<idx_t>());
+            if (it_object != this->line_map.end()){
+                it_object->second->codes[const_it["key"].as<std::string>()] = const_it["value"].as<std::string>();
+            }
```

Comment:
Logging can be great, or just use find_or_default, and it'll segfault on accessing to a nullptr